block-producer/src/server/api.rs

```diff
@@ -2,57 +2,161 @@ use std::{net::ToSocketAddrs, sync::Arc, time::Duration};
 
 use anyhow::Result;
 use async_trait::async_trait;
+use miden_crypto::utils::Deserializable;
 use miden_node_proto::{
-    block_producer::api_server, domain::BlockInputs, requests::SubmitProvenTransactionRequest,
+    account_id,
+    block_producer::api_server,
+    conversion::convert,
+    digest,
+    domain::BlockInputs,
+    requests::{
+        ApplyBlockRequest, GetBlockInputsRequest, GetTransactionInputsRequest,
+        SubmitProvenTransactionRequest,
+    },
     responses::SubmitProvenTransactionResponse,
+    store::api_client as store_client,
+};
+use miden_objects::{accounts::AccountId, transaction::ProvenTransaction, Digest};
+use tonic::{
+    transport::{Channel, Server},
+    Status,
 };
-use miden_objects::{accounts::AccountId, Digest};
-use tonic::{transport::Server, Status};
 use tracing::info;
 
 use crate::{
     batch_builder::{DefaultBatchBuilder, DefaultBatchBuilderOptions},
     block::Block,
     block_builder::DefaultBlockBuilder,
     config::BlockProducerConfig,
+    constants::COMPONENT,
     state_view::DefaulStateView,
     store::{ApplyBlock, ApplyBlockError, BlockInputsError, Store, TxInputs, TxInputsError},
     txqueue::{DefaultTransactionQueue, DefaultTransactionQueueOptions, TransactionQueue},
     SharedProvenTx,
 };
 
-struct RpcStore {}
+struct DefaultStore {
+    store: store_client::ApiClient<Channel>,
+}
 
 #[async_trait]
-impl ApplyBlock for RpcStore {
+impl ApplyBlock for DefaultStore {
     async fn apply_block(
         &self,
-        _block: Arc<Block>,
+        block: Arc<Block>,
     ) -> Result<(), ApplyBlockError> {
-        todo!()
+        let request = tonic::Request::new(ApplyBlockRequest {
+            block: Some(block.header.into()),
+            accounts: convert(block.updated_accounts.clone()),
+            nullifiers: convert(block.produced_nullifiers.clone()),
+            notes: convert(block.created_notes.clone()),
+        });
+
+        let _ = self
+            .store
+            .clone()
+            .apply_block(request)
+            .await
+            .map_err(|status| ApplyBlockError::GrpcClientError(status.message().to_string()))?;
+
+        Ok(())
     }
 }
 
 #[async_trait]
-impl Store for RpcStore {
+impl Store for DefaultStore {
     async fn get_tx_inputs(
         &self,
-        _proven_tx: SharedProvenTx,
+        proven_tx: SharedProvenTx,
     ) -> Result<TxInputs, TxInputsError> {
-        todo!()
+        let request = tonic::Request::new(GetTransactionInputsRequest {
+            account_ids: vec![proven_tx.account_id().into()],
+            nullifiers: proven_tx
+                .consumed_notes()
+                .iter()
+                .map(|note| note.nullifier().into())
+                .collect(),
+        });
```

Comment:
I don't remember why we made `GetTransactionInputRequest` take more than one account ID as input (and I can't come up with a good reason for this). Let's create an issue to change the `GetTransactionInputs` endpoint appropriately.

Reply:
Maybe to support future multi-account transactions? 

Created #103

Reply:
https://github.com/0xPolygonMiden/miden-node/issues/54#issuecomment-1786167092

it was for multi-account transactions.